src/Npgsql/NpgsqlDataAdapter.cs

```diff
@@ -137,6 +139,13 @@ protected override void OnRowUpdating(RowUpdatingEventArgs value)
             get => (NpgsqlCommand?)base.InsertCommand;
             set => base.InsertCommand = value;
         }
+
+        // Temporary implementation, waiting for official support in System.Data via https://github.com/dotnet/runtime/issues/22109
+        internal Task<int> FillAsync(DataTable dataTable)
```

Comment:
This basically provides an async-over-sync implementation for the GetSchemaAsync APIs - which is a bit useless... In fact, the default implementation for GetSchemaAsync in System.Data already does that, so we don't have to implement anything.

What you did before could be fine - just provide our own internal Fill method, which accepts a standard async flag, and fills in the DataTable from the reader.

Reply:
I have reworked that part, tell me whether it is closer to what you have in mind...